resources/hooks/local/pre-push

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+remote="$1"
+
+while read local_ref local_sha remote_ref remote_sha
+do
+    # Get the branch name
+    local_ref=$(basename $local_ref)
+    break
+done
+    
+# Fetch the GIT diff and format it as command input:
+DIFF=$(git diff --name-only "$remote"/"$local_ref"..HEAD --oneline)
```

Comment:
I have not looked at the full PR and comments, but I have some experience with pre-push commands and I have a few suggestions.

I think it will be better to pass the 4 variables `local_ref`, `local_sha`, `remote_ref` and `remote_sha` to PHP instead of doing a diff and passing this data. Let's handle the logic on the PHP side. Passing the diff makes it impossible to retrieve basic information about the current state of the branch on the PHP end which is possibly needed for certain tasks to run on pre-push. Some examples:

* If `local_ref` contains the string `(delete)` we are currently deleting the current branch. This will mean there is still a diff but we don't need to do any coding standards checks because the branch is being deleted.
* If `local_sha` contains the SHA `0000000000000000000000000000000000000000` then we do not have a local branch, meaning we are in a detached HEAD state.
* If `remote_sha` is `0000000000000000000000000000000000000000` then this is a brand new branch. There will not be a diff. We can use this information to perform a full coding standards check.

Also, this `git diff` command is not bullet proof, it assumes that the developer is pushing their currently checked out branch, but this is not necessarily the case. For example, if you have switched to a new branch and realized you have not yet pushed your previous branch, you can do this command:

```
$ git push my-previous-branch:my-previous-branch
```

In this case the diff will be made between the HEAD of the current branch and the remote of the previous branch, this has not the intended result. A better version would be:

```
$ git diff-tree --no-commit-id --name-only -r '$local_sha' '$remote_sha'
```